ibis/formats/pyarrow.py

```diff
@@ -243,15 +244,30 @@ def convert_scalar(cls, scalar: pa.Scalar, dtype: dt.DataType) -> pa.Scalar:
             return scalar
 
     @classmethod
-    def convert_column(cls, column: pa.Array, dtype: dt.DataType) -> pa.Array:
+    def convert_column(
+        cls, column: pa.Array | pd.Series, dtype: dt.DataType
+    ) -> pa.Array:
+        if isinstance(column, pd.Series):
+            if dtype.is_uuid():
```

Comment:
I don't love this special casing here. we could do doing something like 
`target_pd_type = PandasType.from_ibis(PyarrowType.to_ibis(PyarrowType.from_ibis(dtype)))`
which would eval to

```
PandasType.from_ibis(PyarrowType.to_ibis(PyarrowType.from_ibis(dt.UUID)))
PandasType.from_ibis(PyarrowType.to_ibis(pa.string))
PandasType.from_ibis(dt.String)
pd.StringType
```
but IDK that is a bit of a brainf$#$%k as well.

IDK

Reply:
sounds good lets keep it as is.